src/services/userToRoomJoiner.js

```diff
@@ -0,0 +1,61 @@
+const { User, Room } = require('../entities')
+
+class UserToRoomJoiner {
+    constructor({ userRepo, roomRepo }) {
+        this.userRepo = userRepo
+        this.roomRepo = roomRepo
+    }
+
+    call({ username, roomTitle }) {
+        try {
+            return this.join({ username, roomTitle })
+        } catch (error) {
+            return { error: error.message || error }
+        }
+    }
+
+    async join({ username, roomTitle }) {
+        const user = await this.getUser({ username })
+        const room = await this.getRoom({ title: roomTitle })
+        await this.addUserToRoom(room, user)
+        return { user, room }
+    }
+
+    async getUser({ username }) {
+        const user = new User({ username })
+        const foundUser = await this.tryFindUserByUsername(user.username)
```

Comment:
Why is it necessary to call the `tryFindUserByUsername` and not just call the `tryFindByUsername`?
I mean the first one returns whatever the second one returns, so why creating the extra step?

Reply:
It is a future changes protection. Never mind